### DISCUSSION TITLE
How to achieve the best mail throughput? 🏎

### DISCUSSION THREAD

**Discussion by zdeneksvarc** [postalserver/postal#2148] [Help with using Postal]
Let me open a discussion on tuning Postal v2 message throughput. Let's take as a reference point probably the most typical installation, i.e. Postal in LXC container with a single IP address without using IP pools.

- What hardware parameters. CPU clock? CPU threads? Memory/cache size?  Anything else?
- What fine tuning. [Scale](https://docs.docker.com/engine/reference/commandline/service_scale/)? If scale, which worker? No problem with duplicate messages? Etc.

Thank you for suggestions.

**Comment by willpower232** [comment]
> probably the most typical installation, i.e. Postal in LXC container

Postal v2 is currently supported as a docker container so LXC wouldn't be the most typical installation IMHO

If you're using the default install of Postal via docker compose then you can use its scale option, I don't believe docker swarm is involved at all.

Postal is fully clusterable as long as all processes are looking at the same database and RabbitMQ queue so you can run as many images as you need.

Your scaling needs will depend entirely upon how you use Postal and how many messages you're sending. At the very least you might want to run a few of the worker process to guarantee that your message queue is consumed quickly.

**Comment by zdeneksvarc** [comment]
I don't consider LXC and Docker to be interchangeable but related. That's why many Docker Compose projects run on top of LXC containers, often within Proxmox VE. OK, thanks for the theoretical scaling assumptions. If someone from practice shares their experience, that will be helpful too.
